python/lsst/summit/utils/guiders/seeing.py

```diff
@@ -0,0 +1,518 @@
+            ``'flux'``, ``'expid'``, and ``'filter'``.
+        expid : `int`
+            Exposure ID to select. Only rows with this expid will be
+            retained.
+        fluxCut : `float`
+            Minimum mean flux required to keep a detector. Default is
+            ``4e3``.
+
+        Raises
+        ------
+        ValueError
+            If no rows with the given ``expid`` are found in ``stars``.
+        ValueError
+            If more than one unique filter is present in the selected
+            exposure.
+        """
+        self.expid: int = expid
+
+        # Early exit if no data
+        mask = stars["expid"].eq(expid)
+        if not mask.any():
+            raise ValueError(f"Stars dataframe has no data with expid={expid}")
+
+        starsFiltered = stars[mask].copy()
+
+        # Wavelength information from LSST bandpass centers (nm).
+        lamMap: dict[str, float] = {
+            "u": 372.4,
+            "g": 480.7,
+            "r": 622.1,
+            "i": 755.9,
+            "z": 868.0,
+            "y": 975.3,
+        }
+        band = starsFiltered["filter"].unique()
+        if len(band) != 1:
+            raise ValueError("cannot have more than one filter per exposure")
+        self.wavelength: float = float(lamMap[str(band[0])])
+
+        # Initialize metrics builder and detrend stars.
+        # These classes/objects are external to this module.
+        dummyNmissingStampPlaceholder = 0
+        metricsBuilder: Any = GuiderMetricsBuilder(
+            starsFiltered, dummyNmissingStampPlaceholder  # type: ignore[name-defined]
+        )
+        self.metrics: Any = metricsBuilder.buildMetrics(self.expid)  # type: ignore[attr-defined]
+        starsDetrended: pd.DataFrame = metricsBuilder.detrendStars()  # type: ignore[attr-defined]
+
+        # Build star list per detector with a flux threshold.
+        self.starList: dict[str, pd.DataFrame] = groupStarsPerDetector(starsDetrended, fluxCut)
+        self.detectors: list[str] = list(self.starList.keys())
+
+    def measureVar(self) -> list[float]:
+        """
+        Compute per-detector variances of corrected centroid motions.
+
+        Requires that detrending has already been applied. Variances are
+        scaled to units of (pix × 100)^2, i.e., (centipix)^2.
+
+        Returns
+        -------
+        out : `list[float]`
+            Flattened list of per-detector variances:
+            ``[vx_det1, vy_det1, vx_det2, vy_det2, ...]``. If a detector
+            was excluded or lacks corrected coordinates, entries are
+            ``NaN``.
+        """
+        out: list[float] = []
+        for _, df in self.starList.items():
+            meanFlux = float(np.nanmean(pd.to_numeric(df["flux"], errors="coerce")))
+            hasCorr = "dxfp_corr" in df and "dyfp_corr" in df
+            if meanFlux > 4e3 and hasCorr:
+                vx = float(np.nanvar(pd.to_numeric(df["dxfp_corr"], errors="coerce"))) * (100.0**2)
```

Comment:
You're converting mm to pixel. Can you keep on mm? Or use a global variable MM_TO_PIXEL?

Reply:
There should a DM way of doing this, but I'd wait for Merlin's comment.

Reply:
Yeah, I don't love that either, but it depends how accurate you need it. If you had a wcs or the transforms you could use them. Johnny, do you know what's available at this point in the code?